src/elfpy/types.py

```diff
@@ -350,24 +350,22 @@ def update_market_state(self, market_state: MarketState) -> None:
             else:
                 setattr(self, key, [val])
 
-    def update_agent_wallet(self, log_index: int, agent: Agent) -> None:
+    def update_agent_wallet(self, agent: Agent) -> None:
         """Update each entry in the SimulationState's copy for the agent wallet state
         by appending to the list for each key, or creating a new key.
 
         Arguments
         ---------
-        log_index : int
-            Some index indicating the log entry, typically the simulation run_trade_number
         agent: Agent
             An instantiated Agent object
         """
-        d_state = [log_index] + list(agent.wallet.state)
-        if hasattr(self, f"agent_{agent.wallet.address}"):
-            agent_state = getattr(self, f"agent_{agent.wallet.address}")
-            agent_state.append(d_state)
-            setattr(self, f"agent_{agent.wallet.address}", agent_state)
-        else:
-            setattr(self, f"agent_{agent.wallet.address}", [d_state])
+        for key, value in agent.wallet.state.items():
```

Comment:
It seems like the wallet state is a well-defined datastructure. What is the benefit of using a dictionary rather than a dataclass?

Reply:
using a dictionary allows for trivial conversion to a pandas dataframe from the simulation state. This is definitely also doable as a dataclass, I'll mark it as a TODO.